src/main/java/ch/njol/skript/expressions/ExprDirection.java

```diff
@@ -60,153 +60,181 @@
+		UP(BlockFace.UP, "up"),
+		DOWN(BlockFace.DOWN, "down"),
+
+		NORTH(BlockFace.NORTH, "north"),
+		SOUTH(BlockFace.SOUTH, "south"),
+		EAST(BlockFace.EAST, "east"),
+		WEST(BlockFace.WEST, "west"),
+
+		// When using `(:blah[blah2])` the `tag` will be `blah` or `blahblah2` depending on if the optional is used
+		// as a workaround for this we'll be adding all possible patterns into a map.
+		NORTH_EAST(BlockFace.NORTH_EAST, "northeast", "north east", "north-east"),
+		NORTH_WEST(BlockFace.NORTH_WEST, "northwest", "north west", "north-west"),
+		SOUTH_EAST(BlockFace.SOUTH_EAST, "southeast", "south east", "south-east"),
+		SOUTH_WEST(BlockFace.SOUTH_WEST, "southwest", "south west", "south-west");
+
+		public static final HashMap<String, DirectionMapping> DIRECTION_NAMES = new HashMap<>();
+
+		static {
+			for (DirectionMapping direction : values()) {
+				for (String tag : direction.tags) {
+					DIRECTION_NAMES.putIfAbsent(tag.toLowerCase(Locale.ROOT), direction);
+				}
+			}
+		}
+
+		private final BlockFace blockFace;
+		private final String[] tags;
+
+		DirectionMapping(BlockFace blockFace, String ...tags) {
+			this.blockFace = blockFace;
+			this.tags = tags;
+		}
+
+		public BlockFace getBlockFace() {
+			return this.blockFace;
+		}
+
+	}
+
 	static {
 		// TODO think about parsing statically & dynamically (also in general)
 		// "at": see LitAt
 		// TODO direction of %location% (from|relative to) %location%
 		Skript.registerExpression(ExprDirection.class, Direction.class, ExpressionType.COMBINED,
-				"[%-number% [(block|met(er|re))[s]] [to the]] (" +
-						NORTH + "¦north[(-| |)(" + (NORTH_EAST ^ NORTH) + "¦east|" + (NORTH_WEST ^ NORTH) + "¦west)][(ward(s|ly|)|er(n|ly|))] [of]" +
-						"|" + SOUTH + "¦south[(-| |)(" + (SOUTH_EAST ^ SOUTH) + "¦east|" + (SOUTH_WEST ^ SOUTH) + "¦west)][(ward(s|ly|)|er(n|ly|))] [of]" +
-						"|(" + EAST + "¦east|" + WEST + "¦west)[(ward(s|ly|)|er(n|ly|))] [of]" +
-						"|" + UP + "¦above|" + UP + "¦over|(" + UP + "¦up|" + DOWN + "¦down)[ward(s|ly|)]|" + DOWN + "¦below|" + DOWN + "¦under[neath]|" + DOWN + "¦beneath" +
-						") [%-direction%]",
-				"[%-number% [(block|met(er|re))[s]]] in [the] (0¦direction|1¦horizontal direction|2¦facing|3¦horizontal facing) of %entity/block% (of|from|)",
-				"[%-number% [(block|met(er|re))[s]]] in %entity/block%'[s] (0¦direction|1¦horizontal direction|2¦facing|3¦horizontal facing) (of|from|)",
-				"[%-number% [(block|met(er|re))[s]]] (0¦in[ ]front [of]|0¦forward[s]|2¦behind|2¦backwards|[to the] (1¦right|-1¦left) [of])",
-				"[%-number% [(block|met(er|re))[s]]] horizontal[ly] (0¦in[ ]front [of]|0¦forward[s]|2¦behind|2¦backwards|to the (1¦right|-1¦left) [of])");
+				"[%-number% [(block|met(er|re))[s]]] [to the] (:(north[[-| ](east|west)]|east|south[[ |-](east|west)]|west))[ward[s|ly]|er(n|ly)] [of] [%-direction%]",
+				"[%-number% [(block|met(er|re))[s]]] [to the] (up:(above|over|up[ward[s|ly]])|down:(down[ward[s|ly]]|below|under[neath]|beneath)) [%-direction%]",
```

Comment:
I don't think tags work like you think they do. This should only ever have the tags `north`, `west`, `south`, `east`, `up`, or `down`

2.8.4:
```
> !send north
[21:18:22 INFO]: executing 'send north'
[21:18:22 INFO]: north
> !send north-west
[21:18:24 INFO]: executing 'send north-west'
[21:18:24 INFO]: west and north
> !send east
[21:18:28 INFO]: executing 'send east'
[21:18:28 INFO]: east
> !send south
[21:18:30 INFO]: executing 'send south'
[21:18:30 INFO]: south
> !send north-east
[21:18:33 INFO]: executing 'send north-east'
[21:18:33 INFO]: east and north
```

nightly:
```
> !send north-east
[21:15:12 INFO]: executing 'send north-east'
[21:15:12 INFO]: north
> !send south
[21:15:19 INFO]: executing 'send south'
[21:15:19 INFO]: south
> !send east
[21:15:22 INFO]: executing 'send east'
[21:15:22 INFO]: east
> !send north-west
[21:15:28 INFO]: executing 'send north-west'
[21:15:28 INFO]: north
```

this pr also needs tests!

Reply:
Oh, I also think that this isn't any more readable, tbh
It should really be split into a lot more patterns if you want to truly make it readable.

Reply:
Alrighty then something I did in the past was very wrong and I'll work on fixing this converted to draft and should be able to finish this after work I do leave now but should be back in the next 6-ish hours shouldn't be long after that